hack/local-up-cluster.sh

```diff
@@ -20,6 +20,14 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 # this as root to allow kubelet to open docker's socket, and to write the test
 # CA in /var/run/kubernetes.
 # Usage: `hack/local-up-cluster.sh`.
+# NOTE: kubelet only works on Linux! It will more or less turn the host
+# into a node ...
+if [[ "$(uname)" != "Linux" && "${START_MODE:-}" != "nokubelet" ]]; then
```

Comment:
/hold

you can't set this and `nokubeproxy` at the same time, and the kubelet startup currently graciously degrades with a message about kubelet not being supported

Reply:
should we gracefully degrade to not running kube-proxy the same as kubelet and print a warning then ..?

Reply:
that seems better than forcing the user to set the START_MODE, as long as we think they're going to see the warning and understand that this won't have pods or services ...

Reply:
auto-detecting and skipping kubelet and kube-proxy makes sense to me

Reply:
No need for double quotes for the strings that are not to expand anything. Better use single quotes (or none).
```shell
if [[ "$(uname)" != 'Linux' && "${START_MODE:-}" != 'nokubelet' ]]; then
```